### ISSUE TITLE
Plex Job Template apiVersion missing

### ISSUE THREAD

**Issue by johnsondnz** [k8s-at-home/charts#951]
## Chart
plex version 4.4.0
common version 2.5.0

## images
Both
- ghcr.io/k8s-at-home/plex:latest
- ghcr.io/linuxserver/plex:latest

## Tasks to Replicate
```
$ helm install -f plex/values.yaml plex-test k8s-at-home/plex -n pvr 
Error: failed pre-install: unable to build kubernetes object for deleting hook plex/templates/common.yaml: unable to recognize "": no matches for kind "Job" in version ""
```

## Cause
The error falls in  common.yaml Job construction not creating a line for apiVersion; Below shows that helm doesn't have the apiVersion on a correct line, k8s can't interoperate the Group/Version to use
```
$ helm template -f plex/values.yaml plex k8s-at-home/plex -n pvr
---
# Source: plex/templates/common.yaml
---apiVersion: batch/v1
kind: Job
metadata:
  name: plex-autopermissions
  labels:
    helm.sh/chart: plex-4.4.0
    app.kubernetes.io/name: plex
    app.kubernetes.io/instance: plex
    app.kubernetes.io/version: "v1.22.0.4163-d8c4875dd"
    app.kubernetes.io/managed-by: Helm
  annotations:
   "helm.sh/hook": pre-install,pre-upgrade
   "helm.sh/hook-weight": "-10"
   "helm.sh/hook-delete-policy": hook-succeeded,hook-failed,before-hook-creation
spec:
  template:
    metadata:
    spec:
      restartPolicy: Never
      containers:
        - name: set-mount-permissions
          image: "alpine:3.3"
          command:
          - /bin/sh
          - -c
          - | 
          #args:
          #
          #securityContext:
          #
          volumeMounts:
          
      volumes:
```

## Values
```
---
image:
  repository: ghcr.io/linuxserver/plex
  pullPolicy: IfNotPresent
  tag: latest

strategy:
  type: Recreate

env:
  TZ: Pacific/Auckland
  PUID: 1000
  PGID: 1000

service:
  port:
    port: 32400

ingress:
  enabled: false

persistence:
  config:
    enabled: false
  transcode:
    enabled: true
    mountPath: /transcode
    storageClass: "managed-nfs-storage"
    accessMode: ReadWriteOnce
    size: 20Gi

hostPathMounts:
  - name: "config"
    enabled: true
    mountPath: "/config"
    hostPath: "/mnt/local_configs"
    subPath: plex
    readOnly: false
  - name: "tvshows"
    enabled: true
    mountPath: "/data/tvshows"
    hostPath: "/mnt/nas1_media/tvshows"
    readOnly: false
  - name: "movies"
    enabled: true
    mountPath: "/data/movies"
    hostPath: "/mnt/nas1_media/movies"
    readOnly: false

affinity:
  nodeAffinity:
    requiredDuringSchedulingIgnoredDuringExecution:
      nodeSelectorTerms:
      - matchExpressions:
        - key: kubernetes.io/hostname
          operator: In
          values:
          - nuc1

podSecurityContext:
  runAsUser: 1000
  runAsGroup: 1000
  fsGroup: 1000
  # Hardware acceleration using an Intel iGPU w/ QuickSync
  # These IDs below should be matched to your `video` and `render` group on the host
  # To obtain those IDs run the following grep statement on the host:
  # $ cat /etc/group | grep "video\|render"
  # video:x:44:
  # render:x:109:
  supplementalGroups:
  - 44
  - 127
```

## Expected
Chart deploys like any other in your great repository :)  I have successfully used your charts combined with linuxserver images on sonarr, radar, jacket, nzbget, nzbhydra2, overseerr, qbittorrent.  This is the first time this error has occured when using `hostPathMounts`.

## Other
Disabling `hostPathMounts` removes the Job manifest and the deployment goes ahead without error.

**Comment by onedr0p** [comment]
This job is actually being removed in v3 of the common library, it was mentioned here
https://github.com/k8s-at-home/library-charts/issues/54

**Comment by johnsondnz** [comment]
Cool, looking forward to v3.

**Comment by bjw-s** [comment]
The changes mentioned by @onedr0p will be implemented when #963 is merged 🙂
